subprojects/base-services/src/main/java/org/gradle/util/internal/GUtil.java

```diff
@@ -571,4 +573,22 @@ public static boolean isSecureUrl(URI url) {
         return !"http".equalsIgnoreCase(scheme);
     }
 
+    /**
+     * Checks the entry name for zip-slip vulnerable sequences.
+     *
+     * @throws IllegalArgumentException if the entry contains vulnerable sequences
+     */
+    public static String safeZipEntryName(String name) {
+        if (isUnsafeZipEntryName(name)) {
+            throw new IllegalArgumentException(format("'%s' is not a safe zip entry name.", name));
+        }
+        return name;
+    }
+
+    public static boolean isUnsafeZipEntryName(String name) {
+        return name.isEmpty()
+            || name.contains("..")
+            || name.startsWith("/")
+            || name.startsWith("\\");
```

Comment:
Is something like `C:\` also an unsafe name on Windows?

Reply:
Thanks, @wolfs. The code is also checking for `:` now.